### PR TITLE
[PyTorch] RFC: existing MHA: can't we fuse the attn_mask addition easily?

### PR DESCRIPTION
**Pull request by swolchok**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72691

Saw a report that this elementwise add is causing overhead. IIUC this is easy to fuse?

Differential Revision: [D34160547](https://our.internmc.facebook.com/intern/diff/D34160547/)

### CODE REVIEW

**Comment by vadimkantorov** [comment]
Maybe related: https://github.com/pytorch/pytorch/issues/39661

**Comment by albanD** [comment]
From the comment in the related PR from @ngimel , this won't actually improve perf most likely (as baddmm is not a single kernel).
Do you have any benchmark to see if that helps?

**Comment by ngimel** [comment]
It might help a little bit, or might not, depending on the sizes, but a longer term plan is to use masked softmax for this.
